### PR TITLE
create tests for mp b2b upselling products

### PR DESCRIPTION
**Pull request by m-alama**
## PR Description

fix missing enabler for b2b tests
create tests for mp b2b upselling products

### CODE REVIEW

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+    
+Cart_contains_product_with_upselling_relation
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:13]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    ...    AND    I send a POST request:    /carts    {"data":{"type":"carts","attributes":{"priceMode":"${gross_mode}","currency":"${currency_code_eur}","store":"${store_de}","name":"Cart-${random}"}}}
+    ...    AND    Save value to a variable:    [data][id]    cart_id
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${concrete_of_product_with_relations_upselling_sku}","quantity": 1}}}
+    ...    AND    Response status code should be:    201
+    When I send a GET request:    /carts/${cart_id}/up-selling-products
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response should contain the array larger than a certain size:    [data]    0
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:22]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    And Each array element of array in response should contain nested property with value:    [data]    type    abstract-products
+    And Response body parameter should be:    [data][0][id]    ${product_related_product_with_upselling_relation_sku}
+    And Response body parameter should not be EMPTY:    [data][0][attributes]
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:25]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/negative.robot [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+Resource    ../../../../../../resources/common/common_api.robot
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+    
+No_cart_is_passing_to_upselling_products_request
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/negative.robot:13]: Done

[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/negative.robot:13]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    And Response body parameter should be:    [data][0][attributes][sku]    ${product_related_product_with_upselling_relation_sku}
+    And Response body parameter should be:    [data][0][attributes][averageRating]    None
+    And Response body parameter should be:    [data][0][attributes][reviewCount]    0
+    And Response body parameter should be:    [data][0][attributes][name]    ${product_related_product_with_upselling_relation_name}
+    And Response body parameter should not be EMPTY:    [data][0][attributes][description]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][attributes]
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:31]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    And Response should contain the array of a certain size:    [data][0][attributes][metaTitle]    0  
+    And Response body parameter should not be EMPTY:    [data][0][attributes][metaKeywords]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][metaDescription]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][attributeNames]
+    And Response body parameter should not be EMPTY:    [data][0][attributes][url]     
+    And Response body has correct self link
+    [Teardown]    Run Keywords    I send a DELETE request:    /carts/${cart_id}
+    ...  AND    Response status code should be:    204
+
+Cart_contains_product_with_upselling_relation_plus_includes
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:67]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${concrete_of_product_with_relations_upselling_sku}","quantity": 2}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${concrete_of_alternative_product_with_relations_upselling_sku}","quantity": 1}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${concrete_available_product_sku}","quantity": 1}}}
+    ...    AND    Response status code should be:    201
+    When I send a GET request:    /carts/${cart_id}/up-selling-products
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
```
[the remark this answers is not in the document]
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+    And Each array element of array in response should contain property with value in:    [included]    [attributes][currencyIsoCode]    ${currency_code_eur}    ${currency_code_dollar}
+    And Response body parameter should be:    [included][3][id]    ${concrete_of_product_with_relations_upselling_NESTED.product_options.OP_insurance.id}
+    And Response body parameter should be:    [included][3][attributes][optionGroupName]    ${concrete_of_product_with_relations_upselling_NESTED.product_options.OP_insurance.option_group_name}
+    And Response body parameter should be:    [included][3][attributes][sku]    ${concrete_of_product_with_relations_upselling_NESTED.product_options.OP_insurance.sku}
+    And Response body parameter should be:    [included][3][attributes][optionName]    ${concrete_of_product_with_relations_upselling_NESTED.product_options.OP_insurance.option_name}
+    And Each array element of array in response should contain property with value in:    [included]    [attributes][currencyIsoCode]    ${currency_code_eur}    ${currency_code_dollar}
+
+
+Cart_contains_product_with_upselling_relation_with_include_product_reviews
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:316]: Please double-check this test, it fails on stage:
image

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Response body parameter should be:    [data][0][id]    ${product_related_product_with_upselling_relation_NESTED.sku}
+    And Response body parameter should be:    [data][0][attributes][sku]    ${product_related_product_with_upselling_relation_NESTED.sku}
+    And Response body parameter should be:    [data][0][attributes][merchantReference]    ${product_related_product_with_upselling_relation_NESTED.merchant_reference}
+    And Response body parameter should be:    [data][0][attributes][averageRating]    None
+    And Response body parameter should be:    [data][0][attributes][reviewCount]    0
+    And Response body parameter should be:    [data][0][attributes][name]    ${product_related_product_with_upselling_relation_NESTED.name}
+    And Response body parameter should be:    [data][0][attributes][description]    ${product_related_product_with_upselling_relation_NESTED.description}
+    And Response body parameter should not be EMPTY:    [data][0][attributes][attributes]
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:28]: Still no need to check empty [attributes]

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 10]
```diff
@@ -0,0 +1,453 @@
...
+    ...    AND    Save value to a variable:    [data][id]    cart_id
+    ...    AND    I send a POST request:    /carts/${cart_id}/items    {"data": {"type": "items","attributes": {"sku": "${concrete_of_product_with_relations_upselling_sku}","quantity": 1}}}
+    ...    AND    Response status code should be:    201
+    When I send a GET request:    /carts/${cart_id}/up-selling-products?include=abstract-product-prices
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Each array element of array in response should contain property:    [data]    relationships
+    And Each array element of array in response should contain nested property:    [data]    [relationships][abstract-product-prices][data]    type
+    And Each array element of array in response should contain nested property:    [data]    [relationships][abstract-product-prices][data]    id
+    And Response body parameter should not be EMPTY:    [included]
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:99]: You check included array elements below

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 11]
```diff
@@ -0,0 +1,453 @@
...
+    ...    AND    Response status code should be:    201
+    When I send a GET request:    /carts/${cart_id}/up-selling-products?include=product-labels
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
+    And Each array element of array in response should contain property:    [included]    attributes
+    And Each array element of array in response should contain property:    [included]    links
+    And Response include should contain certain entity type:    product-labels
+    Each array element of array in response should contain nested property with value:    [included]    [type]    product-labels
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:235]: Please add 'And'

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 12]
```diff
@@ -0,0 +1,453 @@
...
+    ...    AND    Response status code should be:    201
+    When I send a GET request:    /carts/${cart_id}/up-selling-products?include=product-tax-sets
+    Then Response status code should be:    200
+    And Response reason should be:    OK
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
+    And Each array element of array in response should contain property:    [included]    attributes
+    And Each array element of array in response should contain property:    [included]    links
+    And Response include should contain certain entity type:   product-tax-sets
+    Each array element of array in response should contain nested property with value:    [included]    [type]    product-tax-sets
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:260]: Please add 'And'

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot [hunk 13]
```diff
@@ -0,0 +1,453 @@
...
+    And Each array element of array in response should contain nested property:    [data]    [relationships][abstract-product-availabilities][data]    type
+    And Each array element of array in response should contain nested property:    [data]    [relationships][abstract-product-availabilities][data]    id
+    And Each array element of array in response should contain property:    [included]    type
+    And Each array element of array in response should contain property:    [included]    id
+    And Each array element of array in response should contain property:    [included]    attributes
+    And Each array element of array in response should contain property:    [included]    links
+    Each array element of array in response should contain nested property with value:    [included]    [type]    abstract-product-availabilities
+    Each array element of array in response should contain nested property with value:    [included]    [attributes][quantity]    ${stock_is_20}
+    Each array element of array in response should contain nested property with value:    [included]    [attributes][availability]    True
```
Comment (vadym-uskov) [inline tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:215]: Please add 'And'

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 9 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:28]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 10 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:99]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 12 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:260]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 11 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:235]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 13 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:215]: Done

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 8 above)
Comment (m-alama) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:316]: I have checked responses from both env's it seems that product reviews in local mp-b2b ordered by id 
and in staging is NOT, I think it's a bug

Comment (vadym-uskov) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:324]: Do you create the bug-ticket?

**Review by vadym-uskov — COMMENTED** [review]
Need to investigate an issue with failed test

File: tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot (hunk 8 above)
Comment (vadym-uskov) [reply tests/api/mp_b2b/glue/general_product_endpoints/upselling_products/positive.robot:324]: Created bug-ticket https://spryker.atlassian.net/browse/CC-16680

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**